solana/Dockerfile

```diff
@@ -12,7 +12,7 @@ RUN apt-get update && \
     && \
     rm -rf /var/lib/apt/lists/* && \
     rustup component add rustfmt && \
-    rustup default nightly-2022-01-02
+    rustup default nightly-2021-12-03
```

Comment:
is this dangerous to revert our solana builder to an older rust nightly? @kcsongor @hendrikhofstadt ?

Reply:
It's hard to quantify the effects of switching (or even not switching) nightly versions, but what seems like the safest thing to do is to track the compiler version used by the given version of solana we're using. In this case https://github.com/solana-labs/solana/blob/v1.9.4/ci/rust-version.sh#L27 the solana 1.9.4 toolchain seems to be best tested with `2021-12-03`, so I think this is as safe as it gets.

Reply:
Yes, I changed it based on the slack discussion that we should match whatever version solana uses.